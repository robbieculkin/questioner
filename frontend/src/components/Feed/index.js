import React from 'react';
import './index.scss';

const Feed = ({ history }) =>
  <div className='feed'>
    {history.map(message =>
      <div className={`feed-item ${message.fromUser ? 'right' : 'left'}`} key={message.msgId}>
        <div className='text'>
          {message.text}
        </div>
      </div>
    )}
  </div>

export default Feed;