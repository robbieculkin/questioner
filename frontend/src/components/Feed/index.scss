.feed {
  .feed-item {
    &.left {
      text-align: left;
    }
    &.right {
      text-align: right;
    }
    margin-bottom: 10px;
  }
}