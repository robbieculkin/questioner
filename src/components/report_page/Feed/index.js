import React from 'react';
import PropTypes from 'prop-types';
import './index.scss';

const Feed = ({ history }) =>
  <div className='feed'>
    {history.map(message =>
      <div className={`feed-item ${message.fromUser ? 'right' : 'left'}`} key={message.msgId}>
        <div className='text'>
          {message.text}
        </div>
      </div>
    )}
  </div>

Feed.propTypes = {
  history: PropTypes.arrayOf(
    PropTypes.shape({
      msgId: PropTypes.string,
      fromUser: PropTypes.bool,
      text: PropTypes.string
    })
  )
};

Feed.defaultProps = {
  history: []
};

export default Feed;