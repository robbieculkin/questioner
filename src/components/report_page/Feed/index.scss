.feed {
  padding: 40px;

  .feed-item {
    display: flex;
    &.left {
      justify-content: flex-start;
      .text {
        background: #eee;
      }
    }
    &.right {
      justify-content: flex-end;
      .text {
        background: rgb(46, 170, 77);
        // color: #fff;
      }
    }
    .text {
      max-width: 350px;
      padding: 5px 10px;
      margin-bottom: 10px;
      border-radius: 8px;
      word-wrap: break-word;
    }
  }
}