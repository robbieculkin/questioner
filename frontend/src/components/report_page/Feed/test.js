import React from 'react';
import { shallow } from 'enzyme';
import Feed from '.';

describe('Feed', () => {
  it('should render correctly with empty history', () => {
    const feedProps = { history: [] };
    const component = shallow(<Feed {...feedProps} />);
    expect(component).toMatchSnapshot();
  });

  it('should render history correctly', () => {
    const feedProps = {
      history: [
        {
          msgId: 0,
          fromUser: false,
          text: 'Left Text'
        },
        {
          msgId: 1,
          fromUser: true,
          text: 'Right Text'
        }
      ]
    };
    const component = shallow(<Feed {...feedProps} />);
    expect(component).toMatchSnapshot();
  });
});
