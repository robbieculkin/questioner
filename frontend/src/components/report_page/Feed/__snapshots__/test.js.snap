// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feed should render correctly with empty history 1`] = `
<div
  className="feed"
/>
`;

exports[`Feed should render history correctly 1`] = `
<div
  className="feed"
>
  <div
    className="feed-item left"
    key="0"
  >
    <div
      className="text"
    >
      Left Text
    </div>
  </div>
  <div
    className="feed-item right"
    key="1"
  >
    <div
      className="text"
    >
      Right Text
    </div>
  </div>
</div>
`;
